packages/kbn-esql-ast/src/types.ts

```diff
@@ -25,18 +27,24 @@ export interface ESQLLocation {
   max: number;
 }
 
-interface ESQLAstBaseItem {
-  name: string;
+export interface ESQLAstBaseItem<Name = string> {
+  name: Name;
   text: string;
   location: ESQLLocation;
   incomplete: boolean;
 }
 
-export interface ESQLCommand extends ESQLAstBaseItem {
+export interface ESQLCommand<Name = string> extends ESQLAstBaseItem<Name> {
   type: 'command';
   args: ESQLAstItem[];
 }
 
+export interface ESQLAstMetricsCommand extends ESQLCommand<'metrics'> {
```

Comment:
I leave the `args` in `ESQLAstMetricsCommand` for now, as that makes all validation/autocomplete still works, removing `args` makes a lot of changes necessary to validation/autocomplete.